tasks/main.yml

```diff
@@ -1,37 +1,38 @@
 ---
-# tasks file for ansible-role-httpd-openbsd
+# tasks file for ansible-role-httpd_openbsd
 
 - include_vars: "{{ ansible_os_family }}.yml"
 
 - include: "install-{{ ansible_os_family }}.yml"
 
-- name: Create log directory
-  file:
-    path: "{{ httpd-openbsd_log_dir }}"
-    mode: 0755
-    owner: "{{ httpd-openbsd_user }}"
-    group: "{{ httpd-openbsd_group }}"
-    state: directory
-  notify: Restart httpd-openbsd
+- name: Assert items in httpd_openbsd_log_directories has mandatory attributes
+  assert:
+    msg: items in httpd_openbsd_log_directories must have `path` and `state`attriutes defined, and `state` attribute must be either `present` or `absent`
+    that:
+      - "'path' in item"
+      - item.path | length > 0
+      - "'state' in item"
+      - (item.state == 'present') or (item.state == 'absent')
+  with_items: "{{ httpd_openbsd_log_directories }}"
 
-- name: Create db directory
+- name: Create or remove log directories
   file:
-    path: "{{ httpd-openbsd_db_dir }}"
-    mode: 0755
-    owner: "{{ httpd-openbsd_user }}"
-    group: "{{ httpd-openbsd_group }}"
-    state: directory
-  notify: Restart httpd-openbsd
+    path: "{{ item.path }}"
+    mode: "{{ item.mode | oct | default(omit) }}"
```

Comment:
the role is expected to accept a symbolic mode like 'u=rw' as the file module does.
`oct` filter should be modified or it should be documented that the mode must not be a symbolic one.

Reply:
fixed in 79d6e93